src/core/ReactUpdateQueue.js

```diff
@@ -40,20 +41,27 @@ function getInternalInstanceReadyForUpdate(publicInstance, callerName) {
   );
 
   var internalInstance = ReactInstanceMap.get(publicInstance);
-  invariant(
-    internalInstance,
-    '%s(...): Can only update a mounted or mounting component. ' +
-    'This usually means you called %s() on an unmounted ' +
-    'component.',
-    callerName,
-    callerName
-  );
-  invariant(
-    internalInstance !== ReactLifeCycle.currentlyUnmountingInstance,
-    '%s(...): Cannot call %s() on an unmounting component.',
-    callerName,
-    callerName
-  );
```

Comment:
I found this surprising that we removed this. For my own memory... I recall the rationale being that calling it during componentWillUnmount is silly but it is not harmful, unlike calling it AFTER unmount which is indicative of a leak.